### PR TITLE
[stdlib] Collapse PrefixWhileBidirectionalCollection and de-gyb PrefixWhile.swift

### PR DESCRIPTION
**Pull request by airspeedswift**
Basically the same as #14906

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

**Comment by airspeedswift** [comment]
@swift-ci please smoke test compiler performance

File: stdlib/public/core/PrefixWhile.swift [hunk 1]
```diff
@@ -99,70 +100,110 @@ extension LazySequenceProtocol {
...
-  @_inlineable // FIXME(sil-serialize-all)
-  @_versioned // FIXME(sil-serialize-all)
-  internal init(endOf: Base) {
-    self._value = .pastEnd
+extension LazyPrefixWhileCollection {
+  /// A position in the base collection of a `LazyPrefixWhileCollection` or the
+  /// end of that collection.
+  @_fixed_layout // FIXME(sil-serialize-all)
+  @_versioned
+  internal enum _IndexRepresentation {
```
Comment (moiseev) [inline stdlib/public/core/PrefixWhile.swift]: We can reuse the `ClosedRange.Index` enum here. It's essentially the same, just needs to have it's requirements relaxed a little bit (see https://github.com/apple/swift/pull/14288/commits/8b54a33969760c6c5b5b2afd81618890b4503581).

Can you add a `FIXME` please? And maybe a radar/jira.

**Review by moiseev — APPROVED** [review]
:shipit:

File: stdlib/public/core/PrefixWhile.swift (hunk 1 above)
Comment (airspeedswift) [reply stdlib/public/core/PrefixWhile.swift]: OK, I'll merge this and then do that in the other PR, which I need to do another push to.

**Pull request merged**